### ISSUE TITLE
Serious lag issues

### ISSUE THREAD

**Issue by jredd** [Parallels/rq-dashboard#91]
I used rq for handling a queueing system for video rendering in our system. I have about 289 workers from other computers checking into the main controller. Before I ran any jobs through rq-dashboard worked just fine but once I pushed around 112k videos through it, the dashboard just tanked. Even after all the jobs went through the system (rq is a bit of a beast) the dashboard still had issues. Finally after starting and stopping and after a reboot of the system the dashboard finally came back up.

**Comment by karlp** [comment]
Agreed.  I was trying to use it to manage 2200 failed jobs, and it's just rammed at 100% cpu and the page is unusable.  Lots of js errors.  Even with turning the -interval down to 30 seconds it's still wildly far from usable.

**Comment by eoranged** [comment]
@karlp could you please share rq and rq_dashboard versions you have?

**Comment by karlp** [comment]
rq-dashboard:
```
$ rq-dashboard --version
Usage: rq-dashboard [options]

rq-dashboard: error: no such option: --version
$ pip freeze | grep rq
rq==0.4.6
rq-dashboard==0.3.1
```

The redis server itself is (quite!) old, v2.2.12, but I'm not really convinced that's relevant.  I can shell script redis-cli ~faster than rq-dashboard was working

**Comment by gricey432** [comment]
One thing I've noticed is that the dashboard fires off constant ajax requests to the server even if the previous requests haven't come back yet, which overwhelms it pretty fast

**Comment by eoranged** [comment]
@gricey432 Nice catch, Mitchell!
I checked the code for refreshing the jobs and It looks horrifying :)
There are three places with `setInterval`, which is causing this overhelming. Fix is now in master, It will be nice If someone will try It before release.

**Comment by karlp** [comment]
Can't be me I'm afraid, but sounds like fairly clear fixes, yay.  (We've decomissioned the redis server we needed this for)

**Comment by eoranged** [comment]
There are still multiple places where dashboard can be terribly slow, so keeping this open for the time being.
I'm currently working on redesign to catch up with change of internals in RQ 1.x and fix some deep-rooting issues in [separate project](https://github.com/eoranged/squash/), but It's in prototype stage and I'm planning to start experimenting on It actively when I'll finish cleaning up RQ dashboard.

**Comment by gricey432** [comment]
Just a heads up that I've started a rewrite of the client which I was going to submit as a PR in coming weeks. 
Might be worth a chat / new ticket to discuss it with you to avoid any duplicated effort

**Comment by eoranged** [comment]
@gricey432 After a brief review I also think that It will make sense to rewrite UI using React native or Vue. This will allow to make UI a bit cleaner and flexible.
And It will also help to build a mobile app. Because why not :)

I dropped the plan of creating separate project and retuned to old idea of evolutionary development. Also the project is moved to my company's organisation and now I am able to work on It during work hours from time to time.
After #233 I'm thinking to quickly make a few iterations on design to make dashboard look a bit better in terms of structure (so we can display delayed, dependant and complete jobs in more friendly way). After that It will make sense to rewrite the UI to React (or Vue) to avoid doing the same work twice.

What do you think about It?

**Comment by gricey432** [comment]
@eoranged good news about parallels giving you time :)

I don't have any experience with React Native or Vue but I think Vue would be a good option. Personally I wouldn't have any use for a mobile app for this kind of tool, especially since it's hosted internally behind a VPN I can't get on my phone anyway. Other people may have other use cases though. Could make development slower with the big leap away from html to native components, not sure how much I'd be able to contribute if you're going for a mobile app platform.

There are definitely some easy wins to be had on the UI (just looking at one of my instances now and a lot of the boxes go off the white wrapper box, and having the list of queues on every page (I have ~1200) means I have to scroll a long way to even see the content at the bottom. Happy to put some of these minor UI problems into tickets if you like.

**Comment by eoranged** [comment]
@gricey432 Thanks for the feedback!
As for mobile app, It's just a side dish and will be made only If It won't take me more than couple of hours. Or If someone will need It and It won't require big changes in the code.
There are many UI problems right now (and we are in process of fixing them). It will be wonderful If you'll put ones you see into issues. This way we (or anyone else) can fix them at last.
